bufferevent_filter.c

```diff
@@ -441,9 +441,9 @@ be_filter_readcb(struct bufferevent *underlying, void *me_)
 		else
 			state = BEV_NORMAL;
 
-		/* XXXX use return value */
 		res = be_filter_process_input(bevf, state, &processed_any);
-		(void)res;
+		if( res == BEV_ERROR && state == BEV_NORMAL)
+			bufferevent_run_eventcb_(bufev, (BEV_EVENT_ERROR | BEV_EVENT_READING), 0);
```

Comment:
`foo(bar|baz)` or `foo(bar | baz)` instead of `foo((bar | baz))` (if you will look around the code in this file, than you will see that there is first variant).